### DISCUSSION TITLE
Nextjs and .NET core api (azure) - Cookies not being set to browser

### DISCUSSION THREAD

**Discussion by ahmed7am1d** [vercel/next.js#50451] [Help]
### Summary

Hello everyone.
 I have an asp.net core API hosted on azure with the link **https://socialchainapi.azurewebsites.net/**.
 And NextJS app hosted on **https://socialchain-client-api.vercel.app/**
 The problems:

-  I am trying to append cookies from the back-end as below: 
 _**Cookies options:**_
`HttpOnly = true,
        Expires = refreshTokenExpiryDate,
        IsEssential = true,
        Secure = true,
        SameSite = SameSiteMode.None,`
**_Appending the cookies from an HttpPost method :_**
`Response.Cookies.Append("refreshToken", verifyResult.Value.User.RefreshToken, cookieOptionsRefreshToken);
                Response.Cookies.Append("accessToken", verifyResult.Value.AccessToken, cookieOptionsAccessToken);
                var verifyResponse = new VerifyResponse(verifyResult.Value.User.AccountAddress, verifyResult.Value.AccessToken);
                return Ok(verifyResponse);`

- In the front-end (NextJS) I am receiving the cookies with no problem as below, but it is not set in the cookies storage.
**- Network response:** 
![network](https://github.com/vercel/next.js/assets/64076731/4a388ae1-9846-4e58-b52e-39e1af5cd25b)
**- Console (no errors):**
![console](https://github.com/vercel/next.js/assets/64076731/cd849225-3f00-47d2-9456-b707a6d23f12)
**- Browser cookies storage:**
![cookies](https://github.com/vercel/next.js/assets/64076731/5bf0b8e7-2bb0-4171-a68e-cb0ee64b8b35)

I spent more than 4 hours trying to find the problem, but I can't !!!!!!
BTW when hosted locally both working fine !

**Comment by icyJoseph** [comment]
Try setting a domain on the cookie.

Also, in Chrome sometimes, the Requests Panel in Network has a warning sign, which when hovered, says why the cookie is being ignored.

Otherwise, you could debug this by setting a Next.js API route and have your frontend make a request to that endpoint, and in turn send that request to the 3rd party API. You'd need to make sure to pass along the right data in both directions. See if that works.

**Comment by anupong-s** [comment]
@ahmed7am1d
How solve this? I had the same problem.
